website_sale_stock_control/controllers/main.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp.http import request
+from openerp.addons.website_sale.controllers.main import website_sale
+
+
+class WebsiteSale(website_sale):
+
+    def get_attribute_value_ids(self, product):
+        res = super(WebsiteSale, self).get_attribute_value_ids(product)
+        variant_ids = [r[0] for r in res]
+        for r, variant in zip(
+                res, request.env['product.product'].browse(variant_ids)):
+            r.extend([variant.website_qty_available])
+        return res
+
+    def checkout_redirection(self, order):
+        res = super(WebsiteSale, self).checkout_redirection(order=order)
+        order = request.website.sale_get_order(context=request.context)
+        lines = order.order_line.filtered(lambda x: (
+            x.product_id.inventory_availability == 'always' and
+            x.product_uom_qty > x.product_id.website_qty_available)
+        )
+        if lines:
+            return request.redirect("/shop/cart")
```

Comment:
You should show any kind of error message, not just redirect.

Reply:
Ok. Yes i will show a error message with JS

Reply:
@pedrobaeza is not necessary because this controller renders the cart with a icons warning and message "Temporaly out of stock"

Reply:
OK, then please put it a comment before explaining it. As general rule, any thing "weird" should be commented for avoiding code reviewers questions. It's not a question of commenting everything, like "create here the record for...", but things that aren't expected to be should be indicated. In this case:
```python
# redirecting again to the cart will show the text "Temporarily out of stock"
```